src/deprecated.jl

```diff
@@ -1409,3 +1409,310 @@ import Base: empty!
 
 @deprecate deletecols!(df::DataFrame, inds) select!(df, Not(inds))
 @deprecate deletecols(df::DataFrame, inds; copycols::Bool=true) select(df, Not(inds), copycols=copycols)
+
+import Base: getindex
+@deprecate getindex(df::DataFrame, col_ind::ColumnIndex) df[!, col_ind]
+@deprecate getindex(df::DataFrame, col_inds::Union{AbstractVector, Regex, Not}) df[:, col_inds]
+@deprecate getindex(df::DataFrame, ::Colon) df[:, :]
+@deprecate getindex(sdf::SubDataFrame, colind::ColumnIndex) sdf[!, colind]
+@deprecate getindex(sdf::SubDataFrame, colinds::Union{AbstractVector, Regex, Not}) select(sdf, colinds, copycols=false)
+@deprecate getindex(sdf::SubDataFrame, ::Colon) select(sdf, :, copycols=false)
+
+import Base: view
+@deprecate view(adf::AbstractDataFrame, colind::ColumnIndex) view(adf, :, colind)
+@deprecate view(adf::AbstractDataFrame, colinds) view(adf, :, colinds)
+
+import Base: setindex!
+@deprecate setindex!(sdf::SubDataFrame, val::Any, colinds::Any) (sdf[:, colinds] = val; sdf)
+@deprecate setindex!(df::DataFrame, v::AbstractVector, col_ind::ColumnIndex) (df[!, col_ind] = v; df)
+
+# df[SingleColumnIndex] = Single Item (EXPANDS TO NROW(df) if NCOL(df) > 0)
+function Base.setindex!(df::DataFrame, v, col_ind::ColumnIndex)
+    if haskey(index(df), col_ind)
+        Base.depwarn("Implicit broadcasting to an existing column in DataFrame assignment is deprecated." *
+                     "Use an explicit broadcast with df[:, col_ind] .= v.", :setindex!)
```

Comment:
Including the trailing period here might be confusing `df[:, col_ind] .= v.`. In general, I personally like putting syntax expressions in backticks, so this would be: `Use an explicit broadcast with [backtick]df[:, col_ind] .= v[backtick]`

Reply:
Agreed. Fixed